src/cosmic_ray/work_db.py

```diff
@@ -277,8 +278,16 @@ def use_db(path, mode=WorkDB.Mode.create):
       FileNotFoundError: If `mode` is `Mode.open` and `path` does not
         exist.
     """
+    if path is None:
+        tmp_path = tempfile.mkstemp()[1]
+        path = tmp_path
+    else:
+        tmp_path = None
+
     database = WorkDB(path, mode)
     try:
         yield database
     finally:
         database.close()
+        if tmp_path:
```

Comment:
This will not get called if `WorkDB.__init__()` throws an exception.

I think a better solution is to have a context manager that returns either a temp file or the `path` argument. It can use the TempFile context manager internally. This context manager can be wrapped around the entire body.